go/indexer/services/l1/confirmed_headers.go

```diff
@@ -128,47 +128,57 @@ type ConfirmedHeaderSelector struct {
 	cfg HeaderSelectorConfig
 }
 
-func toBlockNumArg(number *big.Int) string {
-	if number == nil {
-		return "latest"
+func HeadersByRange(ctx context.Context, client *rpc.Client, startHeight uint64, count int) ([]*NewHeader, error) {
+	height := startHeight
+	batchElems := make([]rpc.BatchElem, count)
+	for i := 0; i < count; i++ {
+		batchElems[i] = rpc.BatchElem{
+			Method: "eth_getBlockByNumber",
+			Args: []interface{}{
+				util.ToBlockNumArg(new(big.Int).SetUint64(height + uint64(i))),
+				false,
+			},
+			Result: new(NewHeader),
+			Error:  nil,
+		}
 	}
-	pending := big.NewInt(-1)
-	if number.Cmp(pending) == 0 {
-		return "pending"
+
+	if err := client.BatchCallContext(ctx, batchElems); err != nil {
+		return nil, err
 	}
-	return hexutil.EncodeBig(number)
-}
 
-func HeaderByNumber(ctx context.Context, client *rpc.Client, height *big.Int) (*NewHeader, error) {
-	var head *NewHeader
-	err := client.CallContext(ctx, &head, "eth_getBlockByNumber", toBlockNumArg(height), false)
-	if err == nil && head == nil {
-		err = ethereum.NotFound
+	out := make([]*NewHeader, count)
+	for i := 0; i < len(batchElems); i++ {
+		if batchElems[i].Error != nil {
+			return nil, batchElems[i].Error
+		}
+		out[i] = batchElems[i].Result.(*NewHeader)
 	}
-	return head, err
+
+	return out, nil
 }
 
 func (f *ConfirmedHeaderSelector) NewHead(
 	ctx context.Context,
 	lowest uint64,
 	header *types.Header,
 	client *rpc.Client,
-) []*NewHeader {
+) ([]*NewHeader, error) {
 
 	number := header.Number.Uint64()
 	blockHash := header.Hash
 
 	logger.Info("New block", "block", number, "hash", blockHash)
 
 	if number < f.cfg.ConfDepth {
-		return nil
+		return nil, nil
```

Comment:
Returning `nil, nil` seems a bit risky if there aren't explicit checks on in callers of this function, I assume that it does because it previously returned `nil`

Reply:
Yeah, this was the way the interface worked previously.